api/src/main/java/com/codeforcommunity/rest/subrouter/AuthRouter.java

```diff
@@ -88,4 +102,31 @@ private void handlePostNewUser(RoutingContext ctx) {
 
     end(ctx.response(), 201, JsonObject.mapFrom(response).toString());
   }
+
+  private void handleVerifySecretKey(RoutingContext ctx) {
+    String secret = ctx.pathParam("secret_key");
+
+    VerifySecretKeyResponse response = authProcessor.validateSecretKey(secret);
+
+    int status = 200;
+    if (response.getUserId() == -1) {
+      status = 401;
+    }
+
+    end(ctx.response(), status, JsonObject.mapFrom(response).toString());
+  }
+
+  private void createSecretKey(RoutingContext ctx) {
+    int userId = Integer.valueOf(ctx.pathParam("user_id"));
+
+    try {
+      String token = authProcessor.createSecretKey(userId);
+      System.out.println(token);
+    }
+    catch (AuthException e) {
+      e.printStackTrace();
+      e.getMessage();
+    }
+    end(ctx.response(), 418, "Not set up yet");
```

Comment:
Don't leave joke status codes in production

Reply:
NO